rpc/rpcserver/server.go

```diff
@@ -2340,6 +2347,63 @@ func (s *loaderServer) FetchMissingCFilters(ctx context.Context, req *pb.FetchMi
 	return &pb.FetchMissingCFiltersResponse{}, nil
 }
 
+func (s *loaderServer) SpvSync(req *pb.SpvSyncRequest, svr pb.WalletLoaderService_SpvSyncServer) error {
+
+	wallet, ok := s.loader.LoadedWallet()
+	if !ok {
+		return status.Errorf(codes.FailedPrecondition, "Wallet has not been loaded")
+	}
+
+	if req.DiscoverAccounts && len(req.PrivatePassphrase) == 0 {
+		return status.Errorf(codes.InvalidArgument, "private passphrase is required for discovering accounts")
+	}
+
+	if req.DiscoverAccounts {
+		lock := make(chan time.Time, 1)
+		defer func() {
```

Comment:
we should lock when the sync is finished (the first time SyncUpdated is called with true), not when the RPC ends.